docker/serverxml.patch

```diff
@@ -0,0 +1,41 @@
+--- server.xml
++++ serverhttps.xml
+@@ -66,16 +66,16 @@
+          APR (HTTP/AJP) Connector: /docs/apr.html
+          Define a non-SSL/TLS HTTP/1.1 Connector on port 8080
+     -->
++    <!--
+     <Connector port="8080" protocol="HTTP/1.1"
+                connectionTimeout="20000"
+                redirectPort="8443" />
++    -->
+     <!-- A "Connector" using the shared thread pool-->
+-    <!--
+     <Connector executor="tomcatThreadPool"
+                port="8080" protocol="HTTP/1.1"
+                connectionTimeout="20000"
+                redirectPort="8443" />
```

Comment:
If you change the HTTPs port mapped on your host system any redirect from HTTP will still go to 8443 because the server just returns a location header with this;

eg: in Docker Compose if I use 80:8080 / 443:8443 and hit http://localhost my browser will be redirected to 8443 which is not mapped to anything. 

This needs to be dynamic

Reply:
This has been addressed in commit e4936a9

Reply:
Also how to use docker is in commit 33518b2